### PR TITLE
Warn if the user tries to use a plain HTTP server URL

### PR DESCRIPTION
**Pull request by evgeni**
and fail if the URL is neither HTTPS nor HTTP.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -91,6 +91,7 @@ tests/test_playbooks/vars/server.yml:
 
 dist-test: $(MANIFEST)
 	FOREMAN_SERVER_URL=https://foreman.example.test ansible -m $(NAMESPACE).$(NAME).organization -a "username=admin password=changeme name=collectiontest" localhost | grep -q "Failed to connect to Foreman server.*foreman.example.test"
+	FOREMAN_SERVER_URL=http://foreman.example.test ansible -m $(NAMESPACE).$(NAME).organization -a "username=admin password=changeme name=collectiontest" localhost 2>&1| grep -q "You have configured a plain HTTP server URL."
```
Comment (mdellweg) [inline Makefile:94]: This is to show the warning is happening?

Comment (evgeni) [reply Makefile:94]: correct.

**Comment by evgeni** [comment]
The CodeQL warning is interesting, but unrelated.

**Review by mdellweg — APPROVED** [review]

File: tests/test_crud.py [hunk 2]
```diff
@@ -8,6 +8,7 @@
 
 IGNORED_WARNINGS = [
     "Activation Key 'Test Activation Key Copy' already exists.",
+    "You have configured a plain HTTP server URL. All communication will happen unencrypted.",
```
Comment (mdellweg) [inline tests/test_crud.py:11]: Now this is an "expected warning"...

Comment (evgeni) [reply tests/test_crud.py:11]: We use HTTP in the inventory tests, and I am not going to fix that ;)

**Pull request merged**